test/utils/floating_point_utils.hpp

```diff
@@ -66,6 +66,10 @@ struct FP
         return std::isnan(value) ? (as_uint() & mantissa_mask) : 0;
     }
 
+    bool is_canonical_nan() const noexcept { return nan_payload() == canon; }
```

Comment:
Please also update `CanonicalNaN` and `ArithemeticNaN` matcher.

Reply:
I did already.